### DISCUSSION TITLE
Improve aria-label support for columns

### DISCUSSION THREAD

**Discussion by nicolas-c-dev** [TanStack/table#4277] [Ideas] [answered]
Hi, i've noticed that is not possible to add an **aria-label** for every column in the columns definition and it's very important to give **better accessibility**. Sometimes there are abbreviations that screen readers do not interpret, and it gives a bad exprience for those people. 

Could it be simply adding a **new key to ColumnDef type**. Example:

```
const columns: ColumnDef<Record<string, unknown>>[] = [
  {
    header: 'N° Ticket',
    accessorKey: 'ticketNumber',
    ariaLabel: 'Número de ticket'
  },
];
```

**Comment by tannerlinsley** [comment, marked as answer]
Please pass this in column.meta and apply it in your template.

Tanner Linsley

**Reply by nicolas-c-dev** [reply]
Thanks, I can't find an example of use. Could you show me from where to destructure the prop?

**Reply by tannerlinsley** [reply]
columnHelper.accessor(‘firstName’, {
  meta: { myProp: true }
})

then in table:

column.columnDef.meta.myProp
